dspace-api/src/main/java/org/dspace/discovery/IndexClient.java

```diff
@@ -171,24 +89,76 @@ public static void main(String[] args) throws SQLException, IOException, SearchS
+                                                            .getHandleService().resolveToObject(context, param);
             }
             if (dso == null) {
                 throw new IllegalArgumentException("Cannot resolve " + param + " to a DSpace object");
             }
-            log.info("Indexing " + param + " force " + line.hasOption("f"));
+            handler.logInfo("Indexing " + param + " force " + commandLine.hasOption("f"));
             final long startTimeMillis = System.currentTimeMillis();
-            final long count = indexAll(indexer, ContentServiceFactory.getInstance().getItemService(), context, dso);
+            final long count = indexAll(indexer, ContentServiceFactory.getInstance().getItemService(), context,
+                                        dso);
             final long seconds = (System.currentTimeMillis() - startTimeMillis) / 1000;
-            log.info("Indexed " + count + " object" + (count > 1 ? "s" : "") + " in " + seconds + " seconds");
-        } else {
-            log.info("Updating and Cleaning Index");
-            indexer.cleanIndex(line.hasOption("f"));
-            indexer.updateIndex(context, line.hasOption("f"));
-            checkRebuildSpellCheck(line, indexer);
+            handler.logInfo("Indexed " + count + " object" + (count > 1 ? "s" : "") + " in " + seconds + " seconds");
+        } else if (indexClientOptions == IndexClientOptions.UPDATE ||
+            indexClientOptions == IndexClientOptions.UPDATEANDSPELLCHECK) {
+            handler.logInfo("Updating and Cleaning Index");
+            indexer.cleanIndex(false);
+            indexer.updateIndex(context, false);
+            if (indexClientOptions == IndexClientOptions.UPDATEANDSPELLCHECK) {
+                checkRebuildSpellCheck(commandLine, indexer);
+            }
+        } else if (indexClientOptions == IndexClientOptions.FORCEUPDATE ||
+            indexClientOptions == IndexClientOptions.FORCEUPDATEANDSPELLCHECK) {
+            handler.logInfo("Updating and Cleaning Index");
+            indexer.cleanIndex(true);
+            indexer.updateIndex(context, true);
+            if (indexClientOptions == IndexClientOptions.FORCEUPDATEANDSPELLCHECK) {
+                checkRebuildSpellCheck(commandLine, indexer);
+            }
         }
 
-        log.info("Done with indexing");
+        handler.logInfo("Done with indexing");
+    }
+
+    public void setup() throws ParseException {
+        try {
+            context = new Context(Context.Mode.READ_ONLY);
+            context.turnOffAuthorisationSystem();
+        } catch (Exception e) {
+            throw new ParseException("Unable to create a new DSpace Context: " + e.getMessage());
+        }
+
+        indexClientOptions = IndexClientOptions.getIndexClientOption(commandLine);
+    }
+
+    private IndexClient() {
+        Options options = constructOptions();
+        this.options = options;
+    }
+
+    private Options constructOptions() {
```

Comment:
It's a little odd to me that the `contructOptions()` method still is in `IndexClient` even though you moved the Options settings to `IndexClientOptions`.  This means that when you need to add a new option you need to ensure it is added to two places... `constructOptions()` and `IndexClientOptions`.  Is there a way to move these all to the same place to make it easier to maintain?